src/core/surface/completion_queue.h

```diff
@@ -51,6 +52,16 @@ typedef struct grpc_cq_completion {
   gpr_uintptr next;
 } grpc_cq_completion;
 
+/** An alarm associated with a completion queue. */
```

Comment:
This needs to be a public API (in grpc.h).

Would prefer grpc_alarm for the type (yes I realize this needs a global search/replace for the existing type).

API looks right, though exec_ctx args will need to be dropped when going public - create the exec_ctx in her function instead - and api names will need to be adjusted to the new type name.